### ISSUE TITLE
How to use textAlign with UIText?

### ISSUE THREAD

**Issue by Pattentrick** [collinhover/impactplusplus#105]
Hi there,

i am trying to center text inside an UIText entity. Here is my Code so far:

``` javascript
ig.module(
    'game.ui.command-preview'
)
.requires(
    'plusplus.core.config',
    'plusplus.ui.ui-text'
)
.defines(function () {

    var _c  = ig.CONFIG;

    /**
     * Entity that shows a preview of the selected
     * commands e.g "talk to bartender". Part of the Ui.
     *
     * @class
     * @extends ig.UIText
     * @memeberof ig
     */
    ig.CommandPreview = ig.global.CommandPreview = ig.UIText.extend({

        performance: "dynamic",

        text: 'Gehe zu',

        size: {
            x: 320,
            y: 10
        },

        font: new ig.Font( _c.PATH_TO_MEDIA + 'command_preview_font.png' )

    });

});
```

And i am spawning that entity like this:

``` javascript
ig.game.spawnEntity(ig.CommandPreview, 0, 150);
```

Although i set the size property in the code above, the actual size of the entity later on in the game is approximately the width of the text. But even if i set the size porperty in the update method manually to the correct size, the text will not be centered.

I played around with the other UIText/UIElement propertys, but nothing seems to work. So obviously i am doing something very wrong.

Long story short: How do i center text inside an UIText entity? Thanks in advance for helping me out with this problem!

**Comment by collinhover** [comment]
You're not doing anything wrong :) at the moment you can't force the text to be a specific size. It is always the size of the text. There are max width and height properties if you're trying to make sure it doesn't expand too far. Does that help or is there another reason for being specific widths?

**Comment by Pattentrick** [comment]
Yeay! I am doing nothing wrong :)

Hmm ... i used a specific width because i thought i needed that for a proper text align. Like in classic webdev for example, where a html element has a specific width and i add a `text-align:center;`. I thought UIText behaves similar. So if i made my entity 100px wide the text gets centered about 50px.

So there is no build in way to center text on the screen or inside an entity? I could write a method that gets the size of the UIText entity and sets the new position of this entity in relation to the gamescreen. But until now i thought the UIText class could do that for me. 

So what does textAlign do on UIText, if the entity size is always the size of the text? Is this only for multiline purpose, when the next text line should be aligned in the center, or left and right?

Thank you for clarify this for me!

**Comment by collinhover** [comment]
Yes, the text align is really only for multiple lines in the same text element. But centering a UI element in the screen is built into all UI elements. Set `posAsPct === true` and `posPct === { x: 0.5, y: 0.5 }`. If it needs to be in the center of another UI element, check out the `link()` method and the `linkAlign` property.

**Comment by Pattentrick** [comment]
I will check out those methods. Thank you!
